### PR TITLE
unit tests for the plugin code

### PR DESCRIPTION
**Pull request by darecoder**
Added test cases for method `Execute`, `PrintLogs`, `ProcessLogParameters`, `GetDaemonSetPodsList`, `GetDeploymentPodsList`, `GetStatefulsetPodsList`, `GetResourcesPodList`.

### CODE REVIEW

File: pkg/cmd/historical_logs.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams, args []stri
 
 }
 
-func fetchLogs(baseUrl string, logParameters *LogParameters, podname string, podLogsCh chan<- []string) {
+func FetchLogs(baseUrl string, logParameters *LogParameters, podname string, podLogsCh chan<- []string) {
```
Comment (igor-karpukhin) [inline pkg/cmd/historical_logs.go]: Should this method be public?

**Review by igor-karpukhin — COMMENTED** [review]
@darecoder looks good

**Comment by igor-karpukhin** [comment]
@darecoder please rebase this PR and we can merge it. Good job.

File: pkg/cmd/historical_logs.go (hunk 1 above)
Comment (darecoder) [reply pkg/cmd/historical_logs.go]: Yes, it should be public for e2e tests. But for now I have removed that change and added it to e2e tests PR.

**Comment by darecoder** [comment]
> @darecoder please rebase this PR and we can merge it. Good job.

Done with the rebasing, resolved all the merge conflicts!

**Review by igor-karpukhin — APPROVED** [review]
looks good

**Pull request merged**